plugins/providers/bigquery/client.go

```diff
@@ -49,20 +49,20 @@ func newBigQueryClient(projectID string, credentialsJSON []byte) (*bigQueryClien
 // GetDatasets returns all datasets within a project
 func (c *bigQueryClient) GetDatasets(ctx context.Context) ([]*Dataset, error) {
 	var results []*Dataset
-	it := c.client.Datasets(ctx)
-	for {
-		dataset, err := it.Next()
-		if err == iterator.Done {
-			break
-		}
-		if err != nil {
-			return nil, err
-		}
 
-		results = append(results, &Dataset{
-			ProjectID: dataset.ProjectID,
-			DatasetID: dataset.DatasetID,
-		})
+	req := c.apiClient.Datasets.List(c.projectID)
```

Comment:
is it possible for `req` to be `nil` ?

Reply:
`req` is a pointer, but looking into the source code there's no chance it'll be a nil